plugin/pkg/admission/gc/gc_admission_test.go

```diff
@@ -34,40 +34,40 @@ import (
 
 type fakeAuthorizer struct{}
 
-func (fakeAuthorizer) Authorize(a authorizer.Attributes) (bool, string, error) {
+func (fakeAuthorizer) Authorize(a authorizer.Attributes) (authorizer.Decision, string, error) {
 	username := a.GetUser().GetName()
 
 	if username == "non-deleter" {
 		if a.GetVerb() == "delete" {
-			return false, "", nil
+			return authorizer.DecisionNoOpinion, "", nil
```

Comment:
The new rule is "no denies, and at least one allow"?

I think this code needs to test the update for both no opinion and deny; @caesarxuchao maybe can make a suggestion.

Reply:
As far as any client of the authorizer interface is concerned, the rule is as it always has been… someone must allow. Whether this authorizer stub returned deny or noopinion would make no difference to this test… false -> no-opinion was the behavior-preserving transformation made to all existing code

Reply:
I think explanation here makes sense.

Reply:
My claim is that, while the current [code](https://github.com/kubernetes/kubernetes/pull/53273/files#diff-e2ee3b815cd8240d9aec1244ab564c76R125) happens behaves correctly if a deny is returned, this test does not test that case and therefore it could break in the future.